pilot/pkg/model/push_context.go

```diff
@@ -846,6 +850,19 @@ func (ps *PushContext) SubsetToLabels(proxy *Proxy, subsetName string, hostname
 	return nil
 }
 
+// NamespaceUpdatedByPeerAuthn returns whether a namespace is affected due to a PeerAuthenticationPolicy change.
+func (ps *PushContext) NamespaceUpdatedByPeerAuthn(namespace string, req *PushRequest) bool {
+	if req.NamespaceUpdatedByPeerAuthn == nil {
+		return false
+	}
+	_, rootUpdated := req.NamespaceUpdatedByPeerAuthn[ps.Mesh.RootNamespace]
```

Comment:
[nit] You may join with the next if. E.g
```
if _, rootUpdated :=  req.NamespaceUpdatedByPeerAuthn[ps.Mesh.RootNamespace]; rootUpdated {
  return true;
}
```

Reply:
done